assignments/prototype-refactor.js

```diff
@@ -1,9 +1,162 @@
-/* 
+/*
+  Object oriented design is commonly used in video games.  For this part of the assignment you will be implementing several constructor functions with their correct inheritance hierarchy.
 
-Prototype Refactor
+  In this file you will be creating three constructor functions: GameObject, CharacterStats, Humanoid.  
 
-1. Copy and paste your code or the solution from yesterday
+  At the bottom of this file are 3 objects that all end up inheriting from Humanoid.  Use the objects at the bottom of the page to test your constructor functions.
+  
+  Each constructor function has unique properties and methods that are defined in their block comments below:
+*/
+  
+/*
+  === GameObject ===
+  * createdAt
+  * name
+  * dimensions (These represent the character's size in the video game)
+  * destroy() // prototype method that returns: `${this.name} was removed from the game.`
+*/
+class GameObject
 
-2. Your goal is to refactor all of this code to use ES6 Classes. The console.log() statements should still return what is expected of them.
+{
+	constructor(attributes){
+	this.createdAt = attributes.createdAt
+	this.dimensions = attributes.dimensions
+}
 
-*/
+	destroy(){
+
+  	return(this.name +' was removed from the game')
+}
+}
+
+  //Update Refactor 
+
+
+  /*
+    === CharacterStats ===
+    * healthPoints
+    * takeDamage() // prototype method -> returns the string '<object name> took damage.'
+    * should inherit destroy() from GameObject's prototype
+  */
+ class CharacterStats extends GameObject
+
+{
+
+	constructor(attributes){
+	super(attributes)
+	
+	this.healthPoints = attributes.healthPoints
+	this.name = attributes.name
+
+
+}
+
+	takeDamage(){
+		return(this.name + 'took damage')
```

Comment:
remember your spacing here. Unless there is a space at the end of `this.name` it would look like this when printed on console: `Sir Mustachiotook damage`